indexing-service/src/main/java/org/apache/druid/indexing/common/actions/SegmentTransactionalAppendAction.java

```diff
@@ -148,65 +146,58 @@ public SegmentPublishResult perform(Task task, TaskActionToolbox toolbox)
+
+      final CriticalAction.Action<SegmentPublishResult> publishAction;
+      final String taskAllocatorId = ((PendingSegmentAllocatingTask) task).getTaskAllocatorId();
+      if (startMetadata == null) {
+        publishAction = () -> toolbox.getIndexerMetadataStorageCoordinator().commitAppendSegments(
+            segments,
+            segmentToReplaceLock,
+            taskAllocatorId,
+            segmentSchemaMapping
+        );
+      } else {
+        publishAction = () -> toolbox.getIndexerMetadataStorageCoordinator().commitAppendSegmentsAndMetadata(
+            segments,
+            segmentToReplaceLock,
+            startMetadata,
+            endMetadata,
+            taskAllocatorId,
+            segmentSchemaMapping
         );
       }
-    }
-
-    TaskLocks.checkLockCoversSegments(task, toolbox.getTaskLockbox(), segments);
 
-    final String datasource = task.getDataSource();
-    final Map<DataSegment, ReplaceTaskLock> segmentToReplaceLock
-        = TaskLocks.findReplaceLocksCoveringSegments(datasource, toolbox.getTaskLockbox(), segments);
-
-    final CriticalAction.Action<SegmentPublishResult> publishAction;
-    final String taskAllocatorId = ((PendingSegmentAllocatingTask) task).getTaskAllocatorId();
-    if (startMetadata == null) {
-      publishAction = () -> toolbox.getIndexerMetadataStorageCoordinator().commitAppendSegments(
-          segments,
-          segmentToReplaceLock,
-          taskAllocatorId,
-          segmentSchemaMapping
-      );
-    } else {
-      publishAction = () -> toolbox.getIndexerMetadataStorageCoordinator().commitAppendSegmentsAndMetadata(
-          segments,
-          segmentToReplaceLock,
-          startMetadata,
-          endMetadata,
-          taskAllocatorId,
-          segmentSchemaMapping
-      );
-    }
-
-    final SegmentPublishResult retVal;
-    try {
+      final SegmentPublishResult retVal;
       retVal = toolbox.getTaskLockbox().doInCriticalSection(
           task,
           segments.stream().map(DataSegment::getInterval).collect(Collectors.toSet()),
           CriticalAction.<SegmentPublishResult>builder()
-              .onValidLocks(publishAction)
-              .onInvalidLocks(
-                  () -> SegmentPublishResult.fail(
-                      "Invalid task locks. Maybe they are revoked by a higher priority task."
-                      + " Please check the overlord log for details."
-                  )
-              )
-              .build()
+                        .onValidLocks(publishAction)
+                        .onInvalidLocks(
+                            () -> SegmentPublishResult.fail(
+                                "Invalid task locks. Maybe they are revoked by a higher priority task."
+                                + " Please check the overlord log for details."
+                            )
+                        )
+                        .build()
```

Comment:
Seems like the original indentation was correct.

Reply:
Reverted